mne/channels/montage.py

```diff
@@ -1292,41 +1327,36 @@ def _set_montage(info, montage, update_ch_names=False, set_dig=True):
                  'left untouched.')
 
     elif isinstance(montage, DigMontage):
+        ch_pos = montage._get_ch_pos()
+        eeg_ref_pos = ch_pos.pop('EEG000', np.zeros(3))
+
+        # This raises based on info being sub/supper set of montage
+        matched_ch_names = _check_ch_names_are_compatible(
+            info_names=set(info['ch_names']),
+            montage_names=set(ch_pos),
+        )
+
+        for name in matched_ch_names:
+            _loc_view = info['chs'][info['ch_names'].index(name)]['loc']
+            _loc_view[:6] = np.concatenate((ch_pos[name], eeg_ref_pos))
 
         if set_dig:
+            # XXX: we need to check backcompat in set_dig=false
+            # XXX: this does not take into account ch_names
+
+            NAMED_KIND = (FIFF.FIFFV_POINT_EEG, FIFF.FIFFV_EEG_CH)
+            is_eeg = [d['kind'] in NAMED_KIND for d in montage.dig]
+            names = iter(montage.ch_names)
+
+            foo = [next(names) if x else None for x in is_eeg]
+            # foo = [None, None, None, 'EEG000', None, None, 'a', 'b', 'c']
```

Comment:
I'm doing something wrong here and I don't understand what is it. The idea is that I would like to get a list of either the ith `ch_name` or `None`.

anyone? @mne-tools/contributors?

Reply:
It should be a generator that either yields none or ch_name (see 
55739fc)